src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -499,7 +499,7 @@ private void Get_CORINFO_SIG_INFO(MethodDesc method, out CORINFO_SIG_INFO sig, b
             Get_CORINFO_SIG_INFO(method.Signature, out sig);
 
             // Does the method have a hidden parameter?
-            if (method.RequiresInstArg() && !isFatFunctionPointer)
+            if (method.RequiresInstArg() && !isFatFunctionPointer && !_compilation.TypeSystemContext.IsSpecialUnboxingThunkTargetMethod(method))
```

Comment:
We could get rid of the special casing here by making `RequiresInstArg` (and likely it's friends too) a virtual method on `MethodDesc`. Might also help when we make `MDArray.Address` method shared generic friendly. I'm conflicted about that.